client/lobby/SelectionTab.cpp

```diff
@@ -797,6 +803,107 @@ std::unordered_set<ResourceID> SelectionTab::getFiles(std::string dirURI, int re
 	return ret;
 }
 
+SelectionTab::CMapInfoTooltipBox::CMapInfoTooltipBox(std::string text, ResourceID resource, ESelectionScreen tabType)
+	: CWindowObject(BORDERED | RCLICK_POPUP)
+{
+	drawPlayerElements = tabType == ESelectionScreen::newGame;
+	renderImage = tabType == ESelectionScreen::newGame;
+
+	OBJ_CONSTRUCTION_CAPTURING_ALL_NO_DISPOSE;
+
+	std::vector<std::shared_ptr<IImage>> images;
+	if(renderImage)
+		images = redrawMinimap(ResourceID(resource.getName(), EResType::MAP), IMAGE_SIZE);
+
+	pos = Rect(0, 0, 2*BORDER + IMAGE_SIZE, 2000);
+	if(renderImage && images.size() > 1)
+		pos.w += IMAGE_SIZE + BORDER;
+
+	label = std::make_shared<CTextBox>(text, Rect(BORDER, BORDER, pos.w-2*BORDER, 350), 0, FONT_MEDIUM, ETextAlignment::CENTER, Colors::WHITE);
+	if(!label->slider)
+		label->resize(label->label->textSize);
+
+	pos.h = BORDER + label->label->textSize.y + BORDER;
+	if(renderImage)
+		pos.h += IMAGE_SIZE + BORDER;
+	backgroundTexture = std::make_shared<CFilledTexture>("DIBOXBCK", pos);
+	updateShadow();
+
+	// TODO: hacky redraw
+	label = std::make_shared<CTextBox>(text, Rect(BORDER, BORDER, pos.w-2*BORDER, 350), 0, FONT_MEDIUM, ETextAlignment::CENTER, Colors::WHITE);
```

Comment:
spaces around binary operators here and in other places (see https://wiki.vcmi.eu/Coding_guidelines)